include/utility/json_utility.hpp

```diff
@@ -903,5 +903,44 @@ inline std::string getServiceName(const nlohmann::json& i_sysCfgJsonObj,
     }
     return std::string{};
 }
+
+/**
+ * @brief An API to check if a FRU is "powerOffOnly"
+ *
+ * Given the system config JSON and VPD FRU path, this API checks if the FRU
+ * VPD can be collected at Chassis Power Off state only.
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ * @param[in] i_vpdFruPath    - EEPROM path.
+ * @return - True if FRU VPD can be collected at Chassis Power Off state only.
+ *           False otherwise
+ */
+inline bool isFruPowerOffOnly(const nlohmann::json& i_sysCfgJsonObj,
+                              const std::string& i_vpdFruPath)
+{
+    if (i_vpdFruPath.empty())
+    {
+        logging::logMessage("Invalid parameters received.");
```

Comment:
Just fur path is checked, so log can't say invalid "parameters".
Also as only one thing is getting checked, it will be good if the log can be specific.